app/assets/javascripts/topology_preview.js

```diff
@@ -1,33 +1,34 @@
 var TopologyPreviewer = (function(){
-  var topologyGraph, svgGroup, data, root, tree, svgHeight, svgWidth;
+  var topologyGraph, presetData, style, svgGroup, data, root, tree, svgHeight, svgWidth,
+      currentStylesheet;
 
   var nodeIds         = 0,
       maxLabelLength  = 20,
-      duration        = 250,
-      endPointCounter = 0,
-      stepSize        = 20,
       ease            = 'cubic-out';
 
   var diagonal = d3.svg.diagonal()
-      .projection(function(d) { return [d.y, d.x]; });
+      .projection(function(d) { return [d.x, d.y]; });
 
   TopologyPreviewer.prototype = {
     preview: function(){
       clear();
 
-      svgHeight = viewerHeight();
-      svgWidth  = viewerWidth();
-      svg       = buildBaseSvg();
-      data      = [graphData()];
-      root      = data[0];
+      currentStylesheet = TopologyStylesheet[style];
```

Comment:
> Also, I think Coffeescript 'converts' it in such a fashion

Indeed, CS gives:

```js
// class TopologyPreviewer
//   constructor: (@topologyGraph, @presetData, @style) ->

var TopologyPreviewer;

TopologyPreviewer = (function() {
  function TopologyPreviewer(topologyGraph, presetData, style) {
    this.topologyGraph = topologyGraph;
    this.presetData = presetData;
    this.style = style;
  }

  return TopologyPreviewer;

})();
```

The key is that all the (instance) variables are assigned in the `TopologyPreviewer` constructor.

> but I'm not a big fan of Coffescript

Perhaps we should look into using ES6 with Babel for future projects? ES6 would solve all these problems for us. :grinning: